File: backend/src/main/java/mx/synectura/nexo_cfdi/shared/domain/user/UserService.java
package mx.synectura.nexo_cfdi.shared.domain.user;

import mx.synectura.nexo_cfdi.shared.domain.user.dto.RegisterUserRequest;
import mx.synectura.nexo_cfdi.shared.domain.user.dto.UserResponse;

public interface UserService {
    UserResponse registerOrUpdate(String microsoftSub, String email, RegisterUserRequest request);
    UserResponse getProfile(String microsoftSub);
}

File: frontend/src/app/core/services/user.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { RegisterUserRequest, UserResponse } from '../models/user.model';
import { environment } from '../../../environments/environment';

@Injectable({ providedIn: 'root' })
export class UserService {
  private readonly http = inject(HttpClient);
  private readonly base = `${environment.apiUrl}/api/v1/users`;

  getProfile(): Observable<UserResponse> {
    return this.http.get<UserResponse>(`${this.base}/profile`);
  }

  saveProfile(request: RegisterUserRequest): Observable<UserResponse> {
    return this.http.post<UserResponse>(`${this.base}/profile`, request);
  }
}
